bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,810 @@
+                    self.config.addr.clone(),
+                    addr.clone(),
+                    sock,
+                    self.conn_tx.clone(),
+                    &self.log,
+                )
+                .await;
+                self.accepted_connections.insert(addr, handle);
+            }
+            Err(err) => {}
+        }
+    }
+
+    // Handle API requests from the `NodeHandle`
+    async fn on_api_request(&mut self, request: NodeApiRequest) {
+        match request {
+            NodeApiRequest::InitRack {
+                rack_uuid,
+                initial_membership,
+                responder,
+            } => {
+                info!(self.log,
+                    "Rack init started";
+                    "rack_uuid" => rack_uuid.to_string()
+                );
+                if self.init_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) = self.fsm.init_rack(
+                    Instant::now().into(),
+                    rack_uuid,
+                    initial_membership,
+                ) {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.init_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::InitLearner { responder } => {
+                if self.init_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) = self.fsm.init_learner(Instant::now().into()) {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.init_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::LoadRackSecret { responder } => {
+                if self.rack_secret_responder.is_some() {
+                    let _ = responder
+                        .send(Err(NodeRequestError::RequestAlreadyPending));
+                    return;
+                }
+                if let Err(err) =
+                    self.fsm.load_rack_secret(Instant::now().into())
+                {
+                    let _ = responder.send(Err(err.into()));
+                } else {
+                    self.rack_secret_responder = Some(responder);
+                    self.deliver_envelopes().await;
+                }
+            }
+            NodeApiRequest::PeerAddresses(peers) => {
+                info!(self.log, "Updated Peer Addresses: {peers:?}");
+                self.manage_connections(peers).await;
+            }
+            NodeApiRequest::GetStatus { responder } => {
+                let status = Status {
```

Comment:
Log here as well.

Reply:
I'm not sure it's useful to log every time a status request is made. Seems like that could make logs more noisy than necessary. This is especially true because status requests are answered immediately unlike the other requests.